src/iminuit/util.py

```diff
@@ -1684,3 +1684,29 @@ def is_positive_definite(m: ArrayLike) -> bool:
             return False
         return True
     return False
+
+
+def is_jupyter() -> bool:
+    try:
+        from IPython import get_ipython
+
+        ip = get_ipython()
+        return ip.has_trait("kernel")
+    except ImportError:
+        return False
+    except AttributeError:
+        return False
```

Comment:
I think that is not necessary, the object returned by get_ipython always has the "has_trait" method.

Reply:
I would have thought so as well, but when calling get_ipython() not in a jupyter notebook / ipython session, it just returns None and that of course has no attribute has_trait.

Reply:
Ok, then please add a comment since this is surprising.

Reply:
https://github.com/scikit-hep/iminuit/blob/3167f0f595ec4f39f2981110a49570e91fc06e38/src/iminuit/util.py#L1697-L1699
I added this comment.